pyuvdata/uvcal/uvcal.py

```diff
@@ -1264,6 +1682,374 @@ def _check_freq_spacing(self, *, raise_errors=True):
+
+        Raises
+        ------
+        ValueError
+            If `cat_name` matches no keys in `phase_center_catalog`.
+        """
+        return uvutils.print_phase_center_info(
+            self.phase_center_catalog,
+            catalog_identifier=catalog_identifier,
+            hms_format=hms_format,
+            return_str=return_str,
+            print_table=print_table,
+        )
+
+    def _update_phase_center_id(self, cat_id, *, new_id=None, reserved_ids=None):
+        """
+        Update a phase center with a new catalog ID number.
+
+        Parameters
+        ----------
+        cat_id : int
+            Current catalog ID of the phase center, which corresponds to a key in the
+            attribute `phase_center_catalog`.
+        new_id : int
+            Optional argument. If supplied, then the method will attempt to use the
+            provided value as the new catalog ID, provided that an existing catalog
+            entry is not already using the same value. If not supplied, then the
+            method will automatically assign a value.
+        reserved_ids : array-like in int
+            Optional argument. An array-like of ints that denotes which ID numbers
+            are already reserved. Useful for when combining two separate catalogs.
+
+        Raises
+        ------
+        ValueError
+            If not using the method on a multi-phase-ctr data set, if there's no entry
+            that matches `cat_name`, or of the value `new_id` is already taken.
+        """
+        new_id = uvutils.generate_new_phase_center_id(
+            phase_center_catalog=self.phase_center_catalog,
+            cat_id=new_id,
+            old_id=cat_id,
+            reserved_ids=reserved_ids,
+        )
+
+        # If new_id is None, it means that the existing ID was fine and needs no update
+        if new_id is not None:
+            self.phase_center_id_array[self.phase_center_id_array == cat_id] = new_id
+            self.phase_center_catalog[new_id] = self.phase_center_catalog.pop(cat_id)
+
+    def _consolidate_phase_center_catalogs(
+        self, *, reference_catalog=None, other=None, ignore_name=False
+    ):
+        """
+        Consolidate phase center catalogs with a reference or another object.
+
+        This is a helper method which updates the phase_center_catalog and related
+        parameters to make this object consistent with a reference catalog or with
+        another object so the second object can be added or concatenated to this object.
+        If both `reference_catalog` and `other` are provided, both this object and the
+        one passed to `other` will have their catalogs updated.
+
+        Parameters
+        ----------
+        reference_catalog : dict
+            A reference catalog to make this object consistent with.
+        other : UVData object
+            A UVData object which self needs to be consistent with because it will be
```

Comment:
UVData -> UVCal

Reply:
This is actually be one where either UVData or UVCal can be used...

Reply:
Adjusted in 766e32646768181b70ddd8d2e576d90ca57c254c.